src/main/scala/net/ripe/rpki/publicationserver/repository/DataFlusher.scala

```diff
@@ -230,11 +231,23 @@ class DataFlusher(conf: AppConfig)(implicit val system: ActorSystem)
     IOStream.string("</notification>", stream)
   }
 
-  def snapshotPath(sessionId: String, serial: Long): Path =
-    Files.createDirectories(Paths.get(conf.rrdpRepositoryPath, sessionId, String.valueOf(serial))).resolve(Rrdp.snapshotFilename)
+  private val random = new scala.util.Random()
 
-  def deltaPath(sessionId: String, serial: Long): Path =
-    Files.createDirectories(Paths.get(conf.rrdpRepositoryPath, sessionId, String.valueOf(serial))).resolve(Rrdp.deltaFilename)
+  private def generateRandomPart: String = {
+    random.shuffle(random.alphanumeric.map(c => c.toLower).take(20).toList).mkString
```

Comment:
Was wondering about that myself 👍

Reply:
This should be ~100 bits (`20*(log(36)/log(2)))`) of entropy, aka 'plenty'.

Reply:
Wouldn't it be easy to guess the pseudo-random generator by the looking at the stream of characters? (I don't believe anyone would do that though).

Reply:
If we want to avoid guessable sequence it is better to use `private val random = new scala.util.Random(new java.security.SecureRandom())`. The default `SecureRandom` is non-blocking as far as I can tell (algorithm is `NativePRNG`, while `SecureRandom.getInstanceStrong()` has algorithm `NativePRNGBlocking`).

Reply:
Yeap, done.